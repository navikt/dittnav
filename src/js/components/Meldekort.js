import React, { Component } from 'react';
import PropTypes from 'prop-types';
import conf from 'js/Config';
import i18n from 'translations/i18n';
import { FormattedMessage as F, injectIntl, intlShape } from 'react-intl';

const fremtidig = (meldekort, getCurrentDate, formatDate) => (meldekort.nesteInnsendingAvMeldekort && getCurrentDate.getTime() < meldekort.nesteInnsendingAvMeldekort
  ? (<F id="meldekort.melding.fremtidig" values={{ dato: formatDate(meldekort.nesteInnsendingAvMeldekort) }} />)
  : null);

const feriedager = meldekort => (meldekort.resterendeFeriedager && meldekort.resterendeFeriedager > 0
  ? (<F id="meldekort.feriedager" values={{ feriedager: meldekort.resterendeFeriedager }} />)
  : null);

const advarsel = risikererTrekk => (risikererTrekk ? (<span><F id="meldekort.trekk" /></span>) : null);

const melding = (next, count, formatDate) => (next ? (
  <F
    id={count === 1 ? 'meldekort.ett' : 'meldekort.flere'}
    values={{
      count, next: next.uke, from: formatDate(next.fra), until: formatDate(next.til),
    }}
  />
) : null);

const trekk = (risikererTrekk, formatDate, next) => (!risikererTrekk ? (<F id="meldekort.info.om.trekk" values={{ dato: formatDate(next.datoForTrekk) }} />) : null);

class Meldekort extends Component {
  render() {
    const { meldekort, intl, getCurrentDate } = this.props;
    const { formatDate } = i18n[intl.locale];
    if (!meldekort) return null;

    const { antallNyeMeldekort: count } = meldekort.nyeMeldekort;
    const risikererTrekk = meldekort.nyeMeldekort.nesteMeldekort && meldekort.nyeMeldekort.nesteMeldekort.risikererTrekk;

    if (count > 0) {
      return (
        <a data-ga="Dittnav/Varsel" className="message clickable meldekort" href={`${conf.dittNav.SERVICES_URL}${conf.ARBEID_PATH}?goto=${conf.MELDEKORT_PATH}`}>
          <span className="icon meldekort-icon" aria-label="alarm-ikon" />
          <span className="texts">
            <span>{fremtidig(meldekort, getCurrentDate, formatDate)} </span>
            <span>{melding(meldekort.nyeMeldekort.nesteMeldekort, count, formatDate)} </span>
            <span>{trekk(risikererTrekk, formatDate, meldekort.nyeMeldekort.nesteMeldekort)} </span>
            <span>{advarsel(risikererTrekk)} </span>
            <p id="meldekort.lenkeTekst">{(count > 1 ? <F id="meldekort.se.oversikt" /> : <F id="meldekort.send" />)}</p>
            <p>{feriedager(meldekort)}</p>
          </span>
        </a>
      );
    }

    if (meldekort.nyeMeldekort.nesteMeldekort) {
      return (
        <div data-ga="Dittnav/Varsel" className="message meldekort">
          <span className="icon meldekort-icon" aria-label="alarm-ikon" />
          <span className="texts">
            <span>{fremtidig(meldekort, getCurrentDate, formatDate)} </span>
            <span>{advarsel(risikererTrekk)} </span>
            <p>{feriedager(meldekort)}</p>
          </span>
        </div>
      );
    }
    return null;
  }
}

const NextCard = PropTypes.shape({
  datoForTrekk: PropTypes.number,
  risikererTrekk: PropTypes.bool,
  uke: PropTypes.string,
  kanSendesFra: PropTypes.number,
  til: PropTypes.number,
  fra: PropTypes.number,
});

const NewCards = PropTypes.shape({
  antallNyeMeldekort: PropTypes.number,
  nesteInnsendingAvMeldekort: PropTypes.number,
  nesteMeldekort: NextCard,
});

export const MeldekortType = PropTypes.shape({ nyeMeldekort: NewCards, remainingHolidays: PropTypes.number });

Meldekort.propTypes = {
  meldekort: MeldekortType,
  getCurrentDate: PropTypes.func,
  intl: intlShape.isRequired, // eslint-disable-line react/no-typos
};

Meldekort.defaultProps = {
  meldekort: null,
  getCurrentDate: () => new Date(),
};

export default injectIntl(Meldekort);
