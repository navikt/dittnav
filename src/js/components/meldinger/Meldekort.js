import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage as F, injectIntl, intlShape } from 'react-intl';
import { Panel } from 'nav-frontend-paneler';
import { Normaltekst, Element } from 'nav-frontend-typografi';
import conf from '../../Config';
import i18n from '../../../translations/i18n';
import { IkonOppgave, LenkepanelMedIkon, createOverskrift } from '../LenkepanelMedIkon';

const fremtidig = (nyeMeldekort, formatDateMonth) => (nyeMeldekort.nesteInnsendingAvMeldekort
  ? (<F id="meldekort.melding.fremtidig" values={{ dato: formatDateMonth(nyeMeldekort.nesteInnsendingAvMeldekort) }} />)
  : null);

const feriedager = meldekort => (meldekort.resterendeFeriedager && meldekort.resterendeFeriedager > 0
  ? (<F id="meldekort.feriedager" values={{ feriedager: meldekort.resterendeFeriedager }} />)
  : null);

const advarsel = risikererTrekk => (risikererTrekk ? (<span><F id="meldekort.trekk" /></span>) : null);

const melding = (next, count, formatDayAndMonth, numberToWord) => (next ? (
  <F
    id={count === 1 ? 'meldekort.ett' : 'meldekort.flere'}
    values={{
      count: numberToWord(count),
      next: next.uke,
      from: formatDayAndMonth(next.fra),
      until: formatDayAndMonth(next.til),
    }}
  />
) : null);

const trekk = (skalViseTrekkdato, formatDateMonth, next) => (skalViseTrekkdato ? (
  <F id="meldekort.info.om.trekk" values={{ dato: formatDateMonth(next.sisteDatoForTrekk) }} />) : null);

class Meldekort extends Component {
  render() {
    const { meldekort, intl } = this.props;
    const { formatDateMonth, formatDayAndMonth, numberToWord } = i18n[intl.locale];
    if (!meldekort) return null;

    const { antallNyeMeldekort: count } = meldekort.nyeMeldekort;
    const risikererTrekk = meldekort.nyeMeldekort.nesteMeldekort && meldekort.nyeMeldekort.nesteMeldekort.risikererTrekk;
    if (count > 0) {
      const overskrift = createOverskrift(
        <>
          <span>{fremtidig(meldekort.nyeMeldekort, formatDateMonth)} </span>
          <span>{melding(meldekort.nyeMeldekort.nesteMeldekort, count, formatDayAndMonth, numberToWord)} </span>
          <span>{trekk(!risikererTrekk, formatDateMonth, meldekort.nyeMeldekort.nesteMeldekort)} </span>
          <span>{advarsel(risikererTrekk)} </span>
        </>,
        'Element',
      );

      const ingress = (
        <>
          <span>{(count > 1 ? <F id="meldekort.se.oversikt" /> : <F id="meldekort.send" />)}. <br /></span>
          {feriedager(meldekort)}
        </>
      );
      return (
        <LenkepanelMedIkon
          className="infoMelding"
          data-ga="Dittnav/Varsel"
          alt="Melding om meldekort"
          overskrift={overskrift}
          ingress={ingress}
          href={`${conf.dittNav.NAV_URL}${conf.MELDEKORT_PATH}`}
        >
          <IkonOppgave />
        </LenkepanelMedIkon>
      );
    }

    if (meldekort.nyeMeldekort.nesteInnsendingAvMeldekort) {
      return (
        <Panel className="meldekortpanel" data-ga="Dittnav/Varsel" border alt="Melding om meldekort">
          <div className="meldekortpanel__ikon">
            <IkonOppgave />
          </div>
          <div className="meldekortpanel__tekst">
            <Element>
              <span>{fremtidig(meldekort.nyeMeldekort, formatDateMonth)} </span>
              <span>{advarsel(risikererTrekk)} </span>
            </Element>
            <Normaltekst>
              {feriedager(meldekort)}
            </Normaltekst>
          </div>
        </Panel>
      );
    }
    return null;
  }
}

const NextCard = PropTypes.shape({
  sisteDatoForTrekk: PropTypes.string,
  risikererTrekk: PropTypes.bool,
  uke: PropTypes.string,
  kanSendesFra: PropTypes.string,
  til: PropTypes.string,
  fra: PropTypes.string,
});

const NewCards = PropTypes.shape({
  antallNyeMeldekort: PropTypes.number,
  nesteInnsendingAvMeldekort: PropTypes.string,
  nesteMeldekort: NextCard,
});

export const MeldekortType = PropTypes.shape({
  nyeMeldekort: NewCards,
  resterendeFeriedager: PropTypes.number,
  etterregistrerteMeldekort: PropTypes.number,
});

Meldekort.propTypes = {
  meldekort: MeldekortType,
  intl: intlShape.isRequired, // eslint-disable-line react/no-typos
};

Meldekort.defaultProps = {
  meldekort: null,
};

export default injectIntl(Meldekort);
