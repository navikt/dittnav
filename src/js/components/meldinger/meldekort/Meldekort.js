import React from 'react';
import { FormattedMessage as F, injectIntl, intlShape } from 'react-intl';
import { Path } from '../../../constants';
import i18n from '../../../../translations/i18n';
import PanelOverskrift from '../../common/PanelOverskrift';
import LenkepanelMedIkon from '../../common/LenkepanelMedIkon';
import { advarsel, feriedager, fremtidig, melding, trekk } from './Meldinger';
import IkonBeskjed from '../../../../assets/IkonBeskjed';
import IkonOppgave from '../../../../assets/IkonOppgave';
import { GoogleAnalyticsAction, GoogleAnalyticsCategory } from '../../../utils/googleAnalytics';
import { buildNavNoUrl } from '../../../utils/api';
import { useMeldekort } from '../../../hooks/usePerson';

const isMeldekortbruker = (meldekort) => (
  meldekort && meldekort.content ? meldekort.content.meldekortbruker : false
);

const Meldekort = ({ intl }) => {
  const [{ data: meldekort, isSuccess }] = useMeldekort();

  if (!isSuccess || !meldekort.content || !isMeldekortbruker(meldekort)) {
    return null;
  }

  const { formatDateMonth, formatDayAndMonth, numberToWord } = i18n[intl.locale];
  const { antallNyeMeldekort } = meldekort.content.nyeMeldekort;
  const risikererTrekk = meldekort.content.nyeMeldekort.nesteMeldekort && meldekort.content.nyeMeldekort.nesteMeldekort.risikererTrekk;

  const overskrift = (klarForInnsending) => (
    klarForInnsending ? (
      <>
        <span>{fremtidig(meldekort.content.nyeMeldekort, formatDateMonth)} </span>
        <span>{melding(meldekort.content.nyeMeldekort.nesteMeldekort, antallNyeMeldekort, formatDayAndMonth, numberToWord)} </span>
        <span>{trekk(!risikererTrekk, formatDateMonth, meldekort.content.nyeMeldekort.nesteMeldekort)} </span>
        <span>{advarsel(risikererTrekk)} </span>
      </>
    ) : (
      <>
        <span>{fremtidig(meldekort.content.nyeMeldekort, formatDateMonth)} </span>
        <span>{advarsel(risikererTrekk)} </span>
      </>
    )
  );

  const ingress = (klarForInnsending) => (
    klarForInnsending ? (
      <>
        <span>{feriedager(meldekort.content)} {(antallNyeMeldekort > 1 ? <F id="meldekort.se.oversikt" /> : <F id="meldekort.send" />)}.</span>
      </>
    ) : (
      <>
        {feriedager(meldekort.content)}
      </>
    )
  );

  if (antallNyeMeldekort > 0) {
    return (
      <LenkepanelMedIkon
        className="infomelding oppgave"
        alt="Melding om meldekort"
        overskrift={<PanelOverskrift overskrift={overskrift(true)} type="Element" />}
        ingress={ingress(true)}
        href={`${buildNavNoUrl(Path.MELDEKORT)}`}
        gaCategory={GoogleAnalyticsCategory.Forside}
        gaAction={GoogleAnalyticsAction.MeldekortKlar}
      >
        <IkonOppgave />
      </LenkepanelMedIkon>
    );
  }

  if (meldekort.content.nyeMeldekort.nesteInnsendingAvMeldekort) {
    return (
      <LenkepanelMedIkon
        className="infomelding meldekort-innsendt"
        alt="Melding om meldekort"
        overskrift={<PanelOverskrift overskrift={overskrift(false)} type="Element" />}
        ingress={ingress(false)}
        href={`${buildNavNoUrl(Path.MELDEKORT)}`}
        gaCategory={GoogleAnalyticsCategory.Forside}
        gaAction={GoogleAnalyticsAction.MeldekortVent}
      >
        <IkonBeskjed />
      </LenkepanelMedIkon>
    );
  }
  return null;
};

Meldekort.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(Meldekort);
