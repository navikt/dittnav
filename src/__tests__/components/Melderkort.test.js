import * as React from 'react';
import Meldekort from 'js/components/Meldekort';
const ReactTestRenderer = require('react-test-renderer');

const getCurrentDate = () => new Date(1531830078487);
const formatDate = date => new Date(date).toLocaleDateString('en-US');

test('basic Meldekort test with no meldekort', () => {
  const component = ReactTestRenderer.create((<Meldekort />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('basic Meldekort test', () => {
  const meldekort = {
    "newCards": {
      "count": 2,
      "nextCard": {
        "datoForTrekk": 1533814941280,
        "risikererTrekk": true,
        "week": "week 42",
        "canBeSubmittedFrom": 1531808013471,
        "until": 1531808093471,
        "from": 1531838093471
      },
      "nextSendingDate": 1541808093471
    },
    "remainingHolidays": 12
  };

  const component = ReactTestRenderer.create((<Meldekort meldekort={meldekort} getCurrentDate={getCurrentDate} formatDate={formatDate} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('basic one Meldekort test', () => {
  const meldekort = {
    "newCards": {
      "count": 1,
      "nextCard": {
        "datoForTrekk": 1533814941280,
        "risikererTrekk": true,
        "week": "week 42",
        "canBeSubmittedFrom": 1531808013471,
        "until": 1531808093471,
        "from": 1531838093471
      },
      "nextSendingDate": 1541808093471
    },
    "remainingHolidays": 12
  };

  const component = ReactTestRenderer.create((<Meldekort meldekort={meldekort} getCurrentDate={getCurrentDate} formatDate={formatDate} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('basic one Meldekort test no risk', () => {
  const meldekort = {
    "newCards": {
      "count": 1,
      "nextCard": {
        "datoForTrekk": 1533814941280,
        "risikererTrekk": false,
        "week": "week 42",
        "canBeSubmittedFrom": 1531808013471,
        "until": 1531808093471,
        "from": 1531838093471
      },
      "nextSendingDate": 1541808093471
    },
    "remainingHolidays": 12
  };

  const component = ReactTestRenderer.create((<Meldekort meldekort={meldekort} getCurrentDate={getCurrentDate} formatDate={formatDate} />));
  expect(component.toJSON()).toMatchSnapshot();
});


test('basic one Meldekort test no risk no remaining holidays', () => {
  const meldekort = {
    "newCards": {
      "count": 1,
      "nextCard": {
        "datoForTrekk": 1533814941280,
        "risikererTrekk": false,
        "week": "week 42",
        "canBeSubmittedFrom": 1531808013471,
        "until": 1531808093471,
        "from": 1531838093471
      },
      "nextSendingDate": 1541808093471
    },
    "remainingHolidays": null
  };

  const component = ReactTestRenderer.create((<Meldekort meldekort={meldekort} getCurrentDate={getCurrentDate} formatDate={formatDate} />));
  expect(component.toJSON()).toMatchSnapshot();
});

test('basic zero Meldekort test', () => {
  const meldekort = {
    "newCards": {
      "count": 0,
      "nextCard": {
        "datoForTrekk": 1533814941280,
        "risikererTrekk": true,
        "week": "week 42",
        "canBeSubmittedFrom": 1531808013471,
        "until": 1531808093471,
        "from": 1531838093471
      },
      "nextSendingDate": 1541808093471
    },
    "remainingHolidays": 12
  };

  const component = ReactTestRenderer.create((<Meldekort meldekort={meldekort} getCurrentDate={getCurrentDate} formatDate={formatDate} />));
  expect(component.toJSON()).toMatchSnapshot();
});