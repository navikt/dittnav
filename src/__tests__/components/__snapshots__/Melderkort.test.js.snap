// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`12 Meldekort test no risk no remaining holidays 1`] = `
<a
  className="message clickable meldekort"
  data-ga="Dittnav/Varsel"
  href="http://localhost:9222/meldekort/"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
      <span>
        Neste meldekort kan sendes fra 11/10/2018
      </span>
       
    </span>
    <span>
      <span>
        Du har tolv meldekort klare for innsending
      </span>
       
    </span>
    <span>
       
    </span>
    <span>
       
    </span>
    <p
      id="meldekort.lenkeTekst"
    >
      <span>
        Oversikt over meldekort
      </span>
    </p>
    <p />
  </span>
</a>
`;

exports[`Meldekort without next card test 1`] = `
<div
  className="message meldekort"
  data-ga="Dittnav/Varsel"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
       
    </span>
    <span>
       
    </span>
    <p>
      <span>
        Du har gjenstående 12 feriedager ved dagpenger.
      </span>
    </p>
  </span>
</div>
`;

exports[`basic Meldekort test 1`] = `
<a
  className="message clickable meldekort"
  data-ga="Dittnav/Varsel"
  href="http://localhost:9222/meldekort/"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
      <span>
        Neste meldekort kan sendes fra 11/10/2018
      </span>
       
    </span>
    <span>
      <span>
        Du har to meldekort klare for innsending
      </span>
       
    </span>
    <span>
       
    </span>
    <span>
      <span>
        <span>
          (Send inn nå, du risikerer trekk!)
        </span>
      </span>
       
    </span>
    <p
      id="meldekort.lenkeTekst"
    >
      <span>
        Oversikt over meldekort
      </span>
    </p>
    <p>
      <span>
        Du har gjenstående 12 feriedager ved dagpenger.
      </span>
    </p>
  </span>
</a>
`;

exports[`basic Meldekort test with no meldekort 1`] = `null`;

exports[`basic one Meldekort test 1`] = `
<a
  className="message clickable meldekort"
  data-ga="Dittnav/Varsel"
  href="http://localhost:9222/meldekort/"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
      <span>
        Neste meldekort kan sendes fra 11/10/2018
      </span>
       
    </span>
    <span>
      <span>
        Du har ett meldekort for uke week 42 (7/17/2018 - 7/17/2018) klart for innsending
      </span>
       
    </span>
    <span>
       
    </span>
    <span>
      <span>
        <span>
          (Send inn nå, du risikerer trekk!)
        </span>
      </span>
       
    </span>
    <p
      id="meldekort.lenkeTekst"
    >
      <span>
        Send meldekort
      </span>
    </p>
    <p>
      <span>
        Du har gjenstående 12 feriedager ved dagpenger.
      </span>
    </p>
  </span>
</a>
`;

exports[`basic one Meldekort test no risk 1`] = `
<a
  className="message clickable meldekort"
  data-ga="Dittnav/Varsel"
  href="http://localhost:9222/meldekort/"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
      <span>
        Neste meldekort kan sendes fra 11/10/2018
      </span>
       
    </span>
    <span>
      <span>
        Du har ett meldekort for uke week 42 (7/17/2018 - 7/17/2018) klart for innsending
      </span>
       
    </span>
    <span>
       
    </span>
    <span>
       
    </span>
    <p
      id="meldekort.lenkeTekst"
    >
      <span>
        Send meldekort
      </span>
    </p>
    <p>
      <span>
        Du har gjenstående 12 feriedager ved dagpenger.
      </span>
    </p>
  </span>
</a>
`;

exports[`basic one Meldekort test no risk no remaining holidays 1`] = `
<a
  className="message clickable meldekort"
  data-ga="Dittnav/Varsel"
  href="http://localhost:9222/meldekort/"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
      <span>
        Neste meldekort kan sendes fra 11/10/2018
      </span>
       
    </span>
    <span>
      <span>
        Du har ett meldekort for uke week 42 (7/17/2018 - 7/17/2018) klart for innsending
      </span>
       
    </span>
    <span>
       
    </span>
    <span>
       
    </span>
    <p
      id="meldekort.lenkeTekst"
    >
      <span>
        Send meldekort
      </span>
    </p>
    <p />
  </span>
</a>
`;

exports[`basic zero Meldekort test 1`] = `
<div
  className="message meldekort"
  data-ga="Dittnav/Varsel"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
      <span>
        Neste meldekort kan sendes fra 11/10/2018
      </span>
       
    </span>
    <span>
      <span>
        <span>
          (Send inn nå, du risikerer trekk!)
        </span>
      </span>
       
    </span>
    <p>
      <span>
        Du har gjenstående 12 feriedager ved dagpenger.
      </span>
    </p>
  </span>
</div>
`;

exports[`more than 13 Meldekort test no risk no remaining holidays 1`] = `
<a
  className="message clickable meldekort"
  data-ga="Dittnav/Varsel"
  href="http://localhost:9222/meldekort/"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
      <span>
        Neste meldekort kan sendes fra 11/10/2018
      </span>
       
    </span>
    <span>
      <span>
        Du har 13 meldekort klare for innsending
      </span>
       
    </span>
    <span>
       
    </span>
    <span>
       
    </span>
    <p
      id="meldekort.lenkeTekst"
    >
      <span>
        Oversikt over meldekort
      </span>
    </p>
    <p />
  </span>
</a>
`;

exports[`nesteInnsendingAvMeldekort is null 1`] = `
<a
  className="message clickable meldekort"
  data-ga="Dittnav/Varsel"
  href="http://localhost:9222/meldekort/"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
       
    </span>
    <span>
      <span>
        Du har ett meldekort for uke week 42 (7/17/2018 - 7/17/2018) klart for innsending
      </span>
       
    </span>
    <span>
       
    </span>
    <span>
       
    </span>
    <p
      id="meldekort.lenkeTekst"
    >
      <span>
        Send meldekort
      </span>
    </p>
    <p>
      <span>
        Du har gjenstående 12 feriedager ved dagpenger.
      </span>
    </p>
  </span>
</a>
`;

exports[`no cards no risk 1`] = `
<div
  className="message meldekort"
  data-ga="Dittnav/Varsel"
>
  <span
    aria-label="alarm-ikon"
    className="icon meldekort-icon"
  />
  <span
    className="texts"
  >
    <span>
      <span>
        Neste meldekort kan sendes fra 11/10/2018
      </span>
       
    </span>
    <span>
       
    </span>
    <p>
      <span>
        Du har gjenstående 12 feriedager ved dagpenger.
      </span>
    </p>
  </span>
</div>
`;
